### PR TITLE
ci: validate PR title against conventional commit rules

### PR DESCRIPTION
**Pull request by dimaqq**
As documented in #1252

- types taken from the doc PR
- scopes were scraped from project PR history

### CODE REVIEW

File: .github/workflows/semantic-pr-check.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+          types: |
+            chore
+            ci
+            docs
+            feat
+            fix
+            perf
+            refactor
+          scopes: |
+            harness
```
Comment (benhoyt) [inline .github/workflows/semantic-pr-check.yaml:30]: We're just going to allow `harness` for now though, right? Per discussion of https://github.com/canonical/operator/pull/1252/files#diff-9769a3f5504af3cd735e08f078b94f827bdd51eacac2e9d80ac9761bc485171fR181

File: .github/workflows/semantic-pr-check.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          types: |
+            chore
+            ci
+            docs
+            feat
+            fix
+            perf
+            refactor
```
Comment (benhoyt) [inline .github/workflows/semantic-pr-check.yaml:28]: Per https://github.com/canonical/operator/pull/1252 I think we want to add `revert` and `test`.

**Review by benhoyt — CHANGES_REQUESTED** [review]
Nice idea, thanks. Couple of changes suggested to the lists, based on #1252.

File: .github/workflows/semantic-pr-check.yaml (hunk 2 above)
Comment (dimaqq) [reply .github/workflows/semantic-pr-check.yaml:28]: Thanks, I've missed that.

File: .github/workflows/semantic-pr-check.yaml (hunk 1 above)
Comment (dimaqq) [reply .github/workflows/semantic-pr-check.yaml:30]: Let me merge this in my fork to see what the meaning of `scopes: [a,b,c]; requireScope: false` is, that is if arbitrary, unlisted scopes are allowed.

Comment (dimaqq) [reply .github/workflows/semantic-pr-check.yaml]: done, added a sane catch-all

**Comment by dimaqq** [comment]
Validated that this works: https://github.com/dimaqq/operator/pull/28

File: .github/workflows/validate-pr-title.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            docs
+            feat
+            fix
+            perf
+            refactor
+            revert
+            test
+          scopes: |
+            harness
+            [a-z]{3,20}
```
Comment (benhoyt) [inline .github/workflows/validate-pr-title.yaml:37]: Shouldn't we prevent any other scope than `harness`? The PR title should leave it off in that case.

Comment (dimaqq) [reply .github/workflows/validate-pr-title.yaml:37]: Let's chat at next standup.
I thought we'd want to allow free-form scope, for:
- outstanding #1148 
- upcoming work like `feat(notice)`
- hypothetical `feat(immutable)` work

I'm happy to go with whatever team decides, this is meant to be pragmatic after all.

Comment (dimaqq) [reply .github/workflows/validate-pr-title.yaml]: Decided at the stand-up: remove free-form scope.

**Review by benhoyt — APPROVED** [review]
Thanks!

**Review by tonyandrewmeyer — APPROVED** [review]
Thanks!

Looks good to me. I did not look at the upstream source to verify that it's not doing anything bad (particularly with the token we are providing) - I'm not super well placed to do that in JS. The upstream repo seems fairly healthy, which is a positive indicator.

**Pull request merged**